### DISCUSSION TITLE
postgres bench - summary with all visualisations

### DISCUSSION THREAD

**Discussion by pc-dok** [TPC-Council/HammerDB#762] [General]
hi all, how i can get in summary over web, output the TPM and also the Metrics, sometimes i get all 3 diagrams, but most of time i have it without metrics, what i make wrong!

regards
franco

**Comment by sm-shaw** [comment]
Hi, the CPU metrics needs to be running before the jobid is created, so the metrics data can be associated with a particular job. So, start the metrics, run the workload, and it should then include a link for the metrics data. This is the same for the timing profiling data as well. This applies to both the GUI and the CLI.

**Comment by pc-dok** [comment]
ok, but for autopilot i can not add metrics from cpu? is that correct?

**Reply by sm-shaw** [reply]
If you start the CPU metrics running before you run a workload then it will also record the metrics for autopilot. Note that Oracle and PostgreSQL have database metrics as well so this will not run under autopilot at present as the web only captures the CPU.

**Comment by pc-dok** [comment]
ok, sorry for the confusing metrics, but yes, i mean under a postgres troc-c workload, the metrics what i can also see live, when i activate it, that i think i have last time also in the summary, but i dont get it anymore there, that is why i asking. sorry for that, i am new in this benchmark tooling!

**Comment by pc-dok** [comment]
ok, the cpu metrics is now showing me in output, i must activate first this metrics agent, and look first if i see something under CPU than, first time connection is failed, so that is why i dont have this output, what are you speaking from postgres metrics than?

**Comment by pc-dok** [comment]
now the output is same as before, but this are only the CPU metrics from the client where i run it, not that what i show, the other metrics with postgres, so wait time wal time and so on, that would be more interessting, can i get this output also in a way of the summary?

**Reply by sm-shaw** [reply]
For the CPU metrics, there is an agent in the agent directory. To get CPU metrics from a remote system rather than the local one (where HammerDB is installed) install HammerDB on the remote system, run the agent and then connect HammerDB to the hostname id/port of the agent. An agent is needed as the data comes from the OS/system and not the database.

This looks like this in the GUI for illustration

<img width="482" alt="Screenshot-2023-03-02-112447" src="https://github.com/user-attachments/assets/8afbb5b7-7e3a-40b9-9f94-bd5763367d91">

and this in web service (on a different system)

![Screenshot-2024-06-27-120355](https://github.com/user-attachments/assets/20672424-abeb-469d-830b-ce1e75aa500d)

(Note CPU utilization can be very useful seeing how a workload is distributed across cores and also for bottlenecks on particualr cores for interrupts)

For PostgreSQL metrics see this blog post here https://www.hammerdb.com/blog/uncategorized/how-to-analyze-postgresql-benchmark-performance-with-hammerdb/ - HammerDB PostgreSQL metrics is based on pgsentinel which takes an active session history approach. This means that you can run multiple workloads and use the metrics window to drag the box in the chart at the top to show different wait events for specific periods of time you are interested. (PostgreSQL metrics comes from the database so does not need an agent). 

![run-3-768x565](https://github.com/user-attachments/assets/72bcaa2d-58c2-46dc-8267-b56d71b69c87)

For stored procedure/function timing/latencies select the Time Profile option and this will produce a chart as below showing the response times for the stored procedures

![Screenshot-at-2023-06-20-16-50-33-768x560](https://github.com/user-attachments/assets/38dd38c1-e70e-4bed-a516-7e99a55eb2f2)

Finally if you run the transaction counter during a workload then you will also get a transaction count chart from the web service, during and after the run

<img width="509" alt="371252131-0a25a0d2-1494-4d4c-9e8f-2d73c36cc955" src="https://github.com/user-attachments/assets/0dca7978-4720-4be1-9515-67f5d4ef3104">

For advanced users I recommend LinuxKI to diagnose what is happening on the system under the database https://github.com/HewlettPackard/LinuxKI

**Comment by pc-dok** [comment]
hi sm-shaw, thanks to that helpfully response. that is now clear for me. the funny thing is, i make this tests in azure, and i change now intro onprem, so my windows client with hammerdb is now a windows server 2019, where i have installed hammerdb and chrome. but i dont see any charts. is there a prereq what i missing. the counts are here, but it generate no chart pictures. in azure on server 22 i have not this issues.

regards
franco

**Reply by pc-dok** [reply]
i think i found the problem. it is onprem, and here all sitting behind a proxy, and local vm is not allowed for internet, so i think java scripts what will need internet, is failing! ok, sorry, than i must find another way!

regards
franco

**Reply by sm-shaw** [reply]
Hi, yes, you are exactly right, this is in the release notes https://www.hammerdb.com/docs/ch01s01.html - the charting uses Apache e-charts so needs an external internet connection for the javascript. You can alternatively in the CLI just get the html for the chart for copying elsewhere.
